platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/StatefulEventHasher.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.hashing;
+
+import com.swirlds.common.crypto.DigestType;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.crypto.HashingOutputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+
+/**
+ * An implementation of {@link EventHasher} that is stateful and thus not safe to use by multiple threads concurrently.
+ */
+public class StatefulEventHasher implements EventHasher {
```

Comment:
It looks like this `StatefulEventHasher` currently only appears in single-use scenarios.

Is the idea that it will eventually be used statefully?

Reply:
yes. I have at least one use case in an upcoming PR, it will be for the event creator. It hashes events when it creates them, and its single threaded, so it can be reused